client/state/domains/suggestions/actions.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	DOMAINS_SUGGESTIONS_RECEIVE,
+	DOMAINS_SUGGESTIONS_REQUEST,
+	DOMAINS_SUGGESTIONS_REQUEST_FAILURE,
+	DOMAINS_SUGGESTIONS_REQUEST_SUCCESS
+} from 'state/action-types';
+
+/**
+ * Returns an action object to be used in signalling that a domains suggestion object
+ * has been received.
+ *
+ * @param   {Array}    suggestions              domain suggestions
+ * @param   {Object}   queryObject              domain suggestions queryObject
+ * @returns {Object}   Action object
+ */
+export function receiveDomainsSuggestions( suggestions, queryObject ) {
+	return {
+		type: DOMAINS_SUGGESTIONS_RECEIVE,
+		queryObject,
+		suggestions
```

Comment:
I've flip-flopped on this myself several times since we've introduced Redux, and I think it comes down to a balance between:

- Adhering to variable naming standards and not necessarily tying state to REST API responses (+mapping)
- Simplifying receiving objects from REST API, having easy to follow code, having data be readily available for consumption (+not mapping)

But as @gwwar mentioned above, many (not all) of these inconsistencies are overcome by use of selectors.

>Components will be using selectors, so this should abstract any naming inconsistencies between php and js.

That said, `getSelectedSite` still returns an object with `is_private` (and other snake-case properties).

I do find the "assembler" functions to be a burden to follow. I wonder if they could just be written as any other action creator, something like `receivePlanFromApi` if they really need to make the distinction of mapping an API object to state object?

It's not something I feel particularly strongly one way or the other, but I tend to lean toward simply receiving API objects into state. Admittedly, it feels a bit untidy to have such inconsistency, but once we start down the path of normalizing objects, it just becomes more cumbersome to work with than warranted by any benefits it might bring.